cache/in-memory/src/lib.rs

```diff
@@ -209,8 +207,6 @@ impl InMemoryCache {
         Ok(self
             .0
             .channels_guild
-            .lock()
-            .await
             .get(&channel_id)
             .map(|x| Arc::clone(&x.data)))
```

Comment:
Wouldn't arc here (and everywhere else in cache) be unnecessary since dashmap gives out immutable references already?

Reply:
I might be misunderstanding something about DashMap, but my understanding is that if any reference (immutable or mutable) is held while a mutable reference is retrieved then it may deadlock. By using Arcs, that's avoided.

Reply:
Correct. That only holds for v4 @Arzte since it may not deadlock.

Reply:
Ahh ok, so how does putting it in an arc get around this? Wouldn't the reference still be around?

Reply:
When something is updated a mutable reference is taken to it and the Arc within it is updated or replaced depending on strong count via `Arc::make_mut`. When the user retrieves something, the cache will get an immutable reference within the map to the Arc and the user will get a clone of that Arc. This way, locks are avoided and the user will have a revision of the resource *at that point in time* when it was the most recent, since if there's an update and the user holds a strong copy of the Arc a new Arc will be made in the cache.

Reply:
Gotcha